app/components/post-search-results-affiliate-item.js

```diff
@@ -7,6 +8,17 @@ export default Component.extend({
 
   classNames: ['post-search-results-affiliate-item'],
 
+  getUnitLink: computed('link', function () {
```

Comment:
Instead of `link` this should listen on both `isInContent` and `affiliateUnit `.